packages/roosterjs-editor-api/lib/format/createLink.ts

```diff
@@ -86,6 +90,9 @@ export default function createLink(
                 /* the selection is not collapsed, use browser execCommand */
                 editor.getDocument().execCommand('createLink', false, normalizedUrl);
                 anchor = getAnchorNodeAtCursor(editor);
+                if (displayText && anchor.textContent != displayText) {
```

Comment:
Wrap to a function?

Reply:
We do duplicate this... I'd be fine with that.